src/WebJobs.Script/Workers/WorkerConcurrencyManager.cs

```diff
@@ -125,8 +129,18 @@ internal async void OnTimer(object sender, System.Timers.ElapsedEventArgs e)
 
                 if (NewWorkerIsRequired(workerStatuses, _addWorkerStopwatch.GetElapsedTime()))
                 {
-                    await _functionInvocationDispatcher.StartWorkerChannel();
-                    _addWorkerStopwatch = ValueStopwatch.StartNew();
+                    if (_functionInvocationDispatcher is RpcFunctionInvocationDispatcher rpcDispatcher)
+                    {
+                        // Check memory
+                        var currentCahannels = (await rpcDispatcher.GetAllWorkerChannelsAsync()).Select(x => x.Process.PrivateMemorySize64);
+                        if (IsEnoughMemory(Process.GetCurrentProcess().PrivateMemorySize64,
+                            currentCahannels,
+                            _memoryLimit))
+                        {
+                            await _functionInvocationDispatcher.StartWorkerChannel();
+                            _addWorkerStopwatch = ValueStopwatch.StartNew();
```

Comment:
Is the stopwatch's purpose here is to measure how long starting a new worker channel takes? If so, Should the stopwatch be starting before calling the `StartWorkerChannel`?

Reply:
The purpose is measure we added a worker last time. We do not want to add new worker too frequently.